app/jobs/importer_base.rb

```diff
@@ -31,59 +31,64 @@ def initialize(customer)
 
   # Overrided in importer_destinations.rb
   def uniq_ref(row)
-    row[:ref]
+    if row[:ref].is_a?(Array)
+      row[:ref].map(&:to_sym)
+    else
+      row[:ref]&.to_sym
+    end
   end
 
   def import(data, name, synchronous, options)
     @synchronous = synchronous
     dests = false
-    refs = []
-
+    refs = Hash.new
     Destination.without_callback(:create, :before, :check_max_destination) do
-      VehicleUsageSet.without_callback(:create, :before, :check_max_vehicle_usage_set) do
-        Customer.transaction do
-          before_import(name, data, options)
-
-          dests = data.each_with_index.collect{ |row, line|
-            # Switch from locale or custom to internal column name in case of csv
-            row = yield(row, line + 1 + (options[:line_shift] || 0))
+      Destination.without_callback(:validation, :before, :update_geocode) do
+        VehicleUsageSet.without_callback(:create, :before, :check_max_vehicle_usage_set) do
+          Customer.transaction do
+            before_import(name, data, options)
+
+            dests = data.map.with_index{ |row, line|
+              # Switch from locale or custom to internal column name in case of csv
+              row = yield(row, line + 1 + (options[:line_shift] || 0))
+
+              next if row.empty? # Skip empty line
+
+              begin
+                if (ref = uniq_ref(row))
+                  if refs.key?(ref)
```

Comment:
If ref is an Array it will not work.

Reply:
That was one of my interrogations. Nevertheless, an Array can be key of a Hash.